### PR TITLE
Add Kafka to osquerybeat+endpoint spec files.

### PR DESCRIPTION
**Pull request by cmacknz**
What the summary says, all of the libbeat based spec files already declare that they support Kafka (which would have only worked for standalone agents). Needed to unblock integration testing of Kafka support for Fleet managed agents.

### CODE REVIEW

File: specs/osquerybeat.spec.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ inputs:
     outputs:
       - elasticsearch
       - logstash
+      - kafka
```
Comment (cmacknz) [inline specs/osquerybeat.spec.yml:16]: @aleksmaus will osquerybeat support Kafka without code changes through libbeat, like the other Beats?

Comment (cmacknz) [reply specs/osquerybeat.spec.yml:16]: Looks like it does, it uses the standard root command that will eventually call the beat `makeOutputFactory` function https://github.com/elastic/beats/blob/c7d59203f2fd7d0a284688ee4c395cb64249f375/libbeat/cmd/instance/beat.go#L377

**Review by pierrehilbert — APPROVED** [review]
LGTM

File: specs/osquerybeat.spec.yml (hunk 1 above)
Comment (aleksmaus) [reply specs/osquerybeat.spec.yml:16]: yeah, osquerybeat relies on libbeat 100%, should work

**Review by aleksmaus — APPROVED** [review]

**Comment by cmacknz** [comment]
```
=== RUN   TestStandaloneUpgradeRetryDownload
    upgrade_test.go:384: Testing Elastic Agent upgrade from 8.10.0-SNAPSHOT to 8.8.0...
    fetcher.go:90: Using existing artifact elastic-agent-8.10.0-SNAPSHOT-linux-arm64.tar.gz
    fixture.go:200: Extracting artifact elastic-agent-8.10.0-SNAPSHOT-linux-arm64.tar.gz to /tmp/TestStandaloneUpgradeRetryDownload2438491427/001
    fixture.go:213: Completed extraction of artifact elastic-agent-8.10.0-SNAPSHOT-linux-arm64.tar.gz to /tmp/TestStandaloneUpgradeRetryDownload2438491427/001
    fixture.go:511: Components were not modified from the fetched artifact
    upgrade_test.go:398: Install the built Agent
    fixture.go:344: >> running agent with: [/tmp/TestStandaloneUpgradeRetryDownload2438491427/001/elastic-agent-8.10.0-SNAPSHOT-linux-arm64/elastic-agent install --force --non-interactive]
    upgrade_test.go:400: Installing in non-interactive mode.Elastic Agent has been successfully installed.
        
    upgrade_test.go:403: Ensure the correct version is running
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent version --yaml]
    upgrade_test.go:502: binary:
          version: 8.10.0
          commit: 0158f7755cd411443587211b3afa89215b7fd201
          build_time: 2023-07-31T13:54:35Z
          snapshot: true
        could not get version. failed to communicate with running daemon: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial unix /run/elastic-agent.sock: connect: no such file or directory"
        Use --binary-only flag to skip trying to retrieve version from running daemon
        
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent version --yaml]
    upgrade_test.go:407: Modify /etc/hosts to simulate transient network error
    upgrade_test.go:413: /etc/hosts modify command:  /usr/bin/sed -i.bak s/localhost/localhost artifacts.elastic.co artifacts-api.elastic.co/g /etc/hosts
    upgrade_test.go:427: Check /etc/hosts command:  /usr/bin/grep artifacts /etc/hosts
    upgrade_test.go:440: Start the Agent upgrade
    upgrade_test.go:452: Check Agent logs for at least two retry messages
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent upgrade 8.8.0]
    upgrade_test.go:462: Find logs command:  /usr/bin/grep download.*retrying --recursive --include *.ndjson /opt/Elastic/Agent/data/elastic-agent-0158f7/logs
    upgrade_test.go:471: [/opt/Elastic/Agent/data/elastic-agent-0158f7/logs/elastic-agent-20230731.ndjson:{"log.level":"warn","@timestamp":"2023-07-31T14:19:58.041Z","log.origin":{"file.name":"upgrade/step_download.go","file.line":184},"message":"unable to download package: 2 errors occurred:\n\t* package '/opt/Elastic/Agent/data/elastic-agent-0158f7/downloads/elastic-agent-8.8.0-linux-arm64.tar.gz' not found: open /opt/Elastic/Agent/data/elastic-agent-0158f7/downloads/elastic-agent-8.8.0-linux-arm64.tar.gz: no such file or directory\n\t* fetching package failed: Get \"https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.8.0-linux-arm64.tar.gz\": dial tcp [::1]:443: connect: connection refused\n\n; retrying (will be retry 1) in 17.831015822s.","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"} /opt/Elastic/Agent/data/elastic-agent-0158f7/logs/elastic-agent-20230731.ndjson:{"log.level":"warn","@timestamp":"2023-07-31T14:20:15.874Z","log.origin":{"file.name":"upgrade/step_download.go","file.line":184},"message":"unable to download package: 2 errors occurred:\n\t* package '/opt/Elastic/Agent/data/elastic-agent-0158f7/downloads/elastic-agent-8.8.0-linux-arm64.tar.gz' not found: open /opt/Elastic/Agent/data/elastic-agent-0158f7/downloads/elastic-agent-8.8.0-linux-arm64.tar.gz: no such file or directory\n\t* fetching package failed: Get \"https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.8.0-linux-arm64.tar.gz\": dial tcp 127.0.0.1:443: connect: connection refused\n\n; retrying (will be retry 2) in 38.104452751s.","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}]
    upgrade_test.go:472: Number of retry messages:  2
    upgrade_test.go:476: Restore /etc/hosts so upgrade can proceed
    upgrade_test.go:481: Waiting for upgrade to finish
    upgrade_test.go:484: Waiting for upgrade watcher to finish running...
    upgrade_test.go:484: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/upgrade_test.go:315
        	            				/home/ubuntu/agent/testing/integration/upgrade_test.go:484
        	Error:      	Condition never satisfied
        	Test:       	TestStandaloneUpgradeRetryDownload
        	Messages:   	agent never removed update marker
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
--- FAIL: TestStandaloneUpgradeRetryDownload (340.32s)
```

**Comment by cmacknz** [comment]
buildkite test it

**Comment by cmacknz** [comment]
https://github.com/elastic/elastic-agent/issues/3155 for the test failure here.

**Comment by cmacknz** [comment]
@Merifyio rebase

**Comment by cmacknz** [comment]
@Mergifyio update

**Review by faec — APPROVED** [review]

**Comment by cmacknz** [comment]
Buildkite is stuck pending but the job actually failed shortly after it started.

```
2023-07-31 15:42:06 EDT | >>> (linux/arm64/ubuntu/22.04[5gib]) Failed for instance 35.224.221.192: cannot continue because stack never became ready
-- | --
  | 2023-07-31 15:42:06 EDT |  
  | 2023-07-31 15:42:19 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Test output (sudo) (stdout): -test.shuffle 1690832539701347841
  | 2023-07-31 15:45:45 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Test output (sudo) (stdout): ok  	github.com/elastic/elastic-agent/testing/integration	206.072s
  | 2023-07-31 15:45:45 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Test output (sudo) (stdout): DONE 1 tests in 259.713s
  | 2023-07-31 15:45:45 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Test output (sudo) (stdout): >> go test: remote-linux-amd64-ubuntu-2204-sudo.integration Test Passed
  | 2023-07-31 15:45:45 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Failed to execute tests on instance: error running sudo tests: failed to fetched test output at $HOME/agent/build/TEST-go-remote-linux-amd64-ubuntu-2204-sudo.integration.out
  | 2023-07-31 15:45:45 EDT | >>> (linux/amd64/ubuntu/22.04[5gid]) Failed for instance 34.29.240.31: failed to execute tests on instance ogc-cj40o8pc1lppmi15gid0-cd46: error running sudo tests: failed to fetched test output at $HOME/agent/build/TEST-go-remote-linux-amd64-ubuntu-2204-sudo.integration.out
```

**Comment by cmacknz** [comment]
buildkite test it

**Comment by cmacknz** [comment]
```
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent status --output json]
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent status --output json]
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent status --output json]
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent status --output json]
    proxy_url_test.go:176: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/proxy_url_test.go:395
        	            				/home/ubuntu/agent/testing/integration/proxy_url_test.go:176
        	Error:      	Condition never satisfied
        	Test:       	TestProxyURL_EnrollProxyAndEmptyProxyInThePolicy
        	Messages:   	want fleet state HEALTHY, got STARTING. agent status: {{    false} 0  [] 0 }
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
    fixture.go:344: >> running agent with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
--- FAIL: TestProxyURL_EnrollProxyAndEmptyProxyInThePolicy (328.36s)
```

Hitting all the test failures with this one.

**Comment by cmacknz** [comment]
buildkite test it

**Comment by cmacknz** [comment]
@Mergifyio update

**Pull request merged**